src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/SchemaGenerator.cs

```diff
@@ -145,9 +145,14 @@ private OpenApiSchema GenerateSchemaForType(Type type, SchemaRepository schemaRe
                 : definitionFactory();
         }
 
-        private OpenApiSchema GeneratePrimitiveSchema(DataContract dataContract)
+        private OpenApiSchema GeneratePrimitiveSchema(DataContract dataContract, SchemaRepository schemaRepository)
         {
-            var schema = new OpenApiSchema
+            if (dataContract.EnumConverter != null)
+            {
+                schemaRepository.RegisterEnumConverter(dataContract.UnderlyingType, dataContract.EnumConverter);
+            }
```

Comment:
I was previously thinking of `SchemaRepository` as something that held all seen schemas. So that it might make sense to store additional information about each of them, e.g. about how to create a value for each schema type in the repository (potentially replacing the existing `OpenApiAnyFactory`).

But anyway, that's not how it works at the moment, since it only stores schemas which are required for `$ref` lookups.

So I agree that adding an enum converter dictionary, one which ended up being for schemas which might not even themselves be in the repository, looks tacked on!

So I've re-done this the way you've suggested, in the latest version of this PR.